### PR TITLE
Implement poll enhancements

### PR DESCRIPTION
**Pull request by cassanof**
# Changes and improvements to poll
- Integration with database
- Using embeds for better UI/UX
- Improved how emojis get detected by the regex and how they are being encoded
- Ability to add descriptions to each emoji by using a comma as separator
- Ability to list all open polls
- Ability to close or delete a poll

### CODE REVIEW

**Review by aiyan — APPROVED** [review]

File: src/commands/poll/subcommands/create.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import ExecutionContext from '../../../ExecutionContext';
+
+const respondWithCountMismatch = (
+  ctx: ExecutionContext,
+  emojiCount: number,
+  descriptionCount: number,
+): Promise<void> => {
+  return ctx.interactionApi.respondWithError(
+    `You supplied ${emojiCount} emojis but ${descriptionCount} descriptions. ` +
+      `There should be exactly 1 description for each emoji, or no descriptions at all.`,
```
Comment (jaysella) [inline src/commands/poll/subcommands/create.ts:15]: Is there a reason these are two separate strings?
```suggestion
    `You supplied ${emojiCount} emojis but ${descriptionCount} descriptions. There should be exactly 1 description for each emoji, or no descriptions at all.`,
```

File: src/commands/poll/subcommands/list.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    const embed: Discord.Embed = {
+      title: 'Your active polls',
+      type: Discord.EmbedType.RICH,
+      color: Discord.Color.PRIMARY,
+      fields: embedFields,
+    };
+    if (embedFields.length < 1) {
+      embed.description = "You don't have any active polls";
+    }
```
Comment (jaysella) [inline src/commands/poll/subcommands/list.ts:32]: Not related to this PR, but we should probably add this logic to other list commands to show a message if there are no results to embed.

File: src/repository/PollRepo.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    guildId: string,
+    userId: string,
+  ): Promise<Cursor<Poll>> => {
+    return this.collection(guildId).find({
+      userId: userId,
+    });
+  };
+
+  deleteById = async (guildId: string, pollId: string): Promise<void> => {
+    this.collection(guildId).deleteMany({ _id: pollId }, function (err) {
```
Comment (jaysella) [inline src/repository/PollRepo.ts]: Should these database actions be `await`?

File: src/commands/poll/subcommands/create.ts (hunk 1 above)
Comment (aiyan) [reply src/commands/poll/subcommands/create.ts:15]: To create a line break in between for code styling. The latter creates a new line 👌

**Review by jaysella — APPROVED** [review]
Looks great!

**Pull request merged**